trust-quorum/src/coordinator_state.rs

```diff
@@ -205,24 +242,300 @@ impl CoordinatorState {
             }
         }
     }
+
+    /// Handle a share response for the last_committed epoch.
+    ///
+    /// If we transition from collecting shares to sending prepare messages,
+    /// we also return our own `PrepareMsg` that must be saved as part of the
+    /// persistent state.
+    pub fn handle_share(
+        &mut self,
+        now: Instant,
+        outbox: &mut Vec<Envelope>,
+        from: PlatformId,
+        epoch: Epoch,
+        share: Share,
+    ) -> Option<PrepareMsg> {
+        match &mut self.op {
+            CoordinatorOperation::CollectShares {
+                last_committed_epoch,
+                last_committed_members,
+                last_committed_threshold,
+                collected_shares,
+                new_shares,
+            } => {
+                // First, perform some validation on the incoming share
+                if *last_committed_epoch != epoch {
```

Comment:
Would recommend creating a new logger with the `last_committed_epoch` set up here so it doesn't get missed in any of the statements below.